source/taurus/enumify/package.d

```diff
@@ -0,0 +1,255 @@
+		&& __traits(compiles, { static if (Args.length > 1) from.std.typecons.Tuple!(Args[1 .. $]) t; })
+	)
+{}
+
+@safe pure nothrow @nogc unittest {
+	assert(!__traits(compiles, Member!()));
+	assert(!__traits(compiles, Member!(int)));
+	assert(!__traits(compiles, Member!("name", "foo")));
+	assert(!__traits(compiles, Member!("name", int, "foo", "bar")));
+	assert(!__traits(compiles, Member!("name", int, "foo", short, "foo")));
+
+	assert( __traits(compiles, Member!("A")));
+	assert( __traits(compiles, Member!("B", int, "name")));
+	assert( __traits(compiles, Member!("C", byte, long)));
+	assert( __traits(compiles, Member!("D", float, char, "name")));
+	assert( __traits(compiles, Member!("E", int, "nameA", from.std.typecons.Tuple!(uint, uint), "nameB")));
+}
+
+/**
+Generate an enum like struct of dynamic values.
+
+Examples:
+---
+// TODO: add Message example
+---
+
+GenAttributes:
+// TODO: add generic generated attr
+*/
+mixin template enumify(from.std.typecons.Flag!"defautToString" defautToString = from.std.typecons.Flag!"defautToString".yes)
+{
+	// takes all attributes used before the mixin call
+	private struct __udahelper() {}
+	// TODO: maybe @disable this();
+
+	import std.format : format;
+	import std.meta : AliasSeq, Filter, staticMap;
+	import std.traits : isType, getUDAs, TemplateArgsOf;
+	import std.typecons : Tuple, tuple;
+	import std.string : toLower;
+
+	import sumtype;
+
+	// Member attribute iteration
+	static foreach (attr; getUDAs!(__udahelper, Member))
+	{
+		static if (TemplateArgsOf!attr.length == 1)
+		{
+			/*
+			Member with name only
+			@Member!("None")
+			---
+			public struct none {
+				string toString() {
+					return "None";
+				}
+			}
+
+			private this(none payload) {
+				handle = payload;
+			}
+
+			public static None() {
+				return typeof(this)(none());
+			}
+			---
+			*/
+
+			// create struct
+			mixin(`public struct `, TemplateArgsOf!attr[0].toLower, ` {
+				string toString() { return "`, TemplateArgsOf!attr[0], `"; }
+			}`);
+
+			// create ctor
+			mixin(`private this()(`, TemplateArgsOf!attr[0].toLower,` payload) {
+				handle = payload;
+			}`);
+
+			// create static method
+			mixin(`public static `, TemplateArgsOf!attr[0], `()() {
+				return typeof(this)(`, TemplateArgsOf!attr[0].toLower,`());
+			}`);
+		}
+		else static if (TemplateArgsOf!attr.length == 2)
+		{
+			/*
+			Member one type
+			@Member!("Some", string)
+			---
+			public struct some {
+				string handle;
+				alias handle this;
+
+				version (D_BetterC) {} else
+				string toString() {
+					return handle.format!"Some(%s)";
+				}
+			}
+
+			private this(some payload) {
+				handle = payload;
+			}
+
+			public static Some(string seq) {
+				return typeof(this)(some(seq));
+			}
+			---
+			*/
+
+			// create struct
+			mixin(`public struct `, TemplateArgsOf!attr[0].toLower, ` {
+				`, TemplateArgsOf!attr[1].stringof, ` handle;
+				alias handle this;
+
+				version (D_BetterC) {} else
+				string toString() {
```

Comment:
Make this a template to make attributes inferred.

Reply:
done